assets/js/src/core/components/tree/components/context-menu/hooks/use-asset-actions.tsx

```diff
@@ -0,0 +1,263 @@
+/**
+* Pimcore
+*
+* This source file is available under two different licenses:
+* - Pimcore Open Core License (POCL)
+* - Pimcore Commercial License (PCL)
+* Full copyright and license information is available in
+* LICENSE.md which is distributed with this source code.
+*
+*  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+*  @license    https://github.com/pimcore/studio-ui-bundle/blob/1.x/LICENSE.md POCL and PCL
+*/
+
+import type { ItemType } from 'antd/es/menu/hooks/useItems'
+import { useTranslation } from 'react-i18next'
+import { Icon } from '@Pimcore/components/icon/icon'
+import React, { useState } from 'react'
+import {
+  api as assetApi,
+  type AssetCreateZipApiResponse,
+  useAssetCreateZipMutation
+} from '@Pimcore/modules/asset/asset-api-slice-enhanced'
+import { invalidatingTags } from '@Pimcore/app/api/pimcore/tags'
+import { useAppDispatch } from '@Pimcore/app/store'
+import { createJob as createDownloadJob } from '@Pimcore/modules/execution-engine/jobs/download/factory'
+import { defaultTopics, topics } from '@Pimcore/modules/execution-engine/topics'
+import { useJobs } from '@Pimcore/modules/execution-engine/hooks/useJobs'
+import { type TreeNodeProps } from '@Pimcore/components/tree/node/tree-node'
+
+export interface NodeAware {
+  node: TreeNodeProps | null
+}
+
+export interface NodeIdAware {
+  nodeId: string | null
+}
+
+export interface AssetContextMenuRename {
+  onClick: () => void
+}
+
+export interface AssetContextMenuDelete {
+  onClick: () => void
+}
+
+export interface AssetContextMenuCopy extends NodeIdAware {}
+export interface AssetContextMenuRefresh extends NodeIdAware {}
+export interface AssetContextMenuDownloadAsZip extends NodeAware {}
+
+export interface UseAssetActionsHookReturn {
+  addFolder: () => ItemType
+  rename: (props: AssetContextMenuRename) => ItemType
+  copy: (props: AssetContextMenuCopy) => ItemType
+  paste: () => ItemType | null
+  cut: () => ItemType
+  remove: (props: AssetContextMenuDelete) => ItemType
+  downloadAsZip: (props: AssetContextMenuDownloadAsZip) => ItemType | null
+  advanced: () => ItemType
+  refresh: (props: AssetContextMenuRefresh) => ItemType
+  requestTranslations: () => ItemType
+  setNodeId: (nodeId: string) => void
+}
+
+export const useAssetActions = (): UseAssetActionsHookReturn => {
+  const { t } = useTranslation()
+  const dispatch = useAppDispatch()
+  const { addJob } = useJobs()
+  const [nodeId, setNodeId] = useState<string | null>(null)
+  const [fetchCreateZip] = useAssetCreateZipMutation()
+
+  const addFolder = (): ItemType => {
+    return {
+      label: t('asset.tree.context-menu.add-folder'),
```

Comment:
should we use "element" as prefix instead of "asset" for the translation keys as we will have the same wordings for all 3 element types?

Reply:
good point: https://github.com/pimcore/studio-ui-bundle/pull/564/commits/13fbb3b79422e51e3ae87c826ae2947b9c56b0f4